### ISSUE TITLE
EMC Storage Cell compatibility

### ISSUE THREAD

**Issue by Mordenkainen** [Mordenkainen/EquivalentEnergistics#72]
Make EMC storage cells compatible with the Transmutation tablet and other EMC powered devices.

Ideally it should be usable wherever a Klein star is.

**Comment by MaPePeR** [comment]
This will not work for ProjectE, because the Tablet does not actually store the EMC. It works like a Vanilla Crafting Table and only works for the player that is looking at it.

**Comment by Mordenkainen** [comment]
Ok, let me see if I am understanding this right. I can't put a charged klein star in a transmutation tablet and use the EMC on it? Instead, if there is any available space in the star, it will absorb whatever EMC it can from the transmutation tablet? Is this correct?

**Comment by MaPePeR** [comment]
Ah i think i totally misunderstood what you were trying to achieve, sorry.

This is what you are looking for, i think: https://github.com/sinkillerj/ProjectE/blob/master/src/main/java/moze_intel/projecte/api/item/IItemEmc.java
It should allow you to implement your own emc-storing-items that act like klein stars.

**Comment by Mordenkainen** [comment]
Yup, that is what I am working with. But I think I see an issue with ProjectE. In https://github.com/sinkillerj/ProjectE/blob/e4e764bde69112b752a98dc5085a7abfbfd3ee13/src/main/java/moze_intel/projecte/gameObjs/container/slots/transmutation/SlotInput.java#L23 an item is only valid for the slot if it has EMC. This is fine for the stars since they have EMC values in addition to being able to store EMC. In the case of my cells, they are IItemEmc items, but have no EMC value of their own, so all checks for if the item has EMC is false. I would suspect the check should be if the item has EMC OR stores EMC.

**Comment by MaPePeR** [comment]
That might be totally possible.

**Comment by Mordenkainen** [comment]
Adding the "question" and "waiting on dependency" labels to this since the fix for the issue really needs to be on the ProjectE side of things. I have submitted PRs for 1.710 and 1.10.2 that fix the issues, but do not know if they will be accepted on these older versions. If they are not I will try to get approval to host versions of PE with these fixes. Otherwise I will have to use some ugly trickery to get it working and I want to avoid that.

The 1.12 version of PE already has the fixes needed in GitHub, but a new release has not been made yet.

**Comment by tomtenberge** [comment]
just give the drives Emc values?

**Comment by Mordenkainen** [comment]
Nope, wouldn't fix the issue.

Some of the PE blocks assume any EMC storage is a Klein Star amd crash if it isn't.

The issue isn't just that it assumes all EMC storage has an EMC value, though that is part of it.

**Comment by MaPePeR** [comment]
> Some of the PE blocks assume any EMC storage is a Klein Star amd crash if it isn't.

That's something that can and should be fixed, though?

**Comment by tomtenberge** [comment]
i agree, that IS a bug, from what i understood from the text here, it was only that the drives themselves didn't have a Emc value... my bad!

**Comment by Mordenkainen** [comment]
@MaPePer

It sure is, but I need it fixed in 1.7 and 1.10 which you guys are not doing releases for.

I already got fixes in for 1.12.

My PRs for 1.7 and 1.10 were closed because you arent doing releases anymore. I had talked to Sin about this, and was told the PRs would be taken if any new release is done.

So, this issue sits until that happens or I decide to do my own builds with the bugs fixed.

**Comment by MaPePeR** [comment]
Ah. Forgot about that.
